documentation/creating_a_script_plan.md

```diff
@@ -0,0 +1,360 @@
+# Wrapping a script in a plan
+
+Running commands and scripts with Bolt is awesome, but at some point you may find yourself wanting
+to do more. Wrapping a script in a plan is a great way to:
+- Make the script discoverable in your project. Your teammates or module users can now find the
+  plan that runs a script by running `bolt plan show`, or `Get-BoltPlan` in PowerShell.
+- Parameterize script arguments. Wrapping your script in a plan let's you enforce types for arguments
+  to your script, and documents those parameters. Users can see how to use your script by running
+  `bolt plan show <plan>`, or `Get-BoltPlan -Name <plan>` in PowerShell.
+- Log messages before or after running the script.
+- Including the script as part of more complex orchestration (for example, automatically running
+  some recovery commands on any targets the script fails on)
+
+**NOTE:** The following example uses a Shell script, but the same workflow applies for any scripting
+language.
+
+Let's say you have a script the gets an updated image for every Docker container running in a
+`docker-compose` project, and you want to turn it into a plan:
+
+_update\_images.sh_
+```
+#!/bin/sh
+for c in `docker-compose ps --services |sort`; do
+  echo "redoing $c"
+  docker-compose rm -s -f $c
+  docker-compose pull $c
+done
+docker-compose up -d --remove-orphans
+```
+
+1. If you haven't done so already, make a new Bolt project in a new directory. We'll call this
+   project 'manage_docker', but you can name it anything.
+
+   _\*nix shell command_
+   ```shell
+   bolt project init
+   ```
+   _PowerShell cmdlet_
+   ```powershell
+   New-BoltProject
+   ```
+2. Make a `scripts/` directory in your Bolt project.
+3. Place the `update_images.sh` script in the `scripts/` directory.
+4. Create a Bolt plan using `bolt plan new --script`.
+   This command will create a Bolt YAML plan that takes a `targets` parameter, runs the script on the
+   targets, and returns the result from the script run. If you'd prefer to create a Puppet language
+   plan instead of YAML, you can run the same command with the `--pp` option.
```

Comment:
```suggestion
   This command creates a Bolt YAML plan that takes a `targets` parameter, runs the script on the
   targets, and returns the result from the script run. If you'd prefer to create a Puppet language
   plan instead of YAML, you can run the same command with the `--pp` option.
```